### PR TITLE
fix: ensure stable imports

### PR DESCRIPTION
**Pull request by jsstevenson**
Found a few edge errors involving circular/inaccessible imports. I'm hoping this PR will resolve all of them. The changes are:

* move core tool functions to a module that doesn't share its name with its package (resolves https://stackoverflow.com/questions/54333865/python-no-module-named-error-package-is-not-a-package). Now `api.py` contains FastAPI stuff and `main.py` contains tool stuff.
* move shared file locations to a `paths.py` module, out of `__init__.py`.
* each module acquires the package logger instance itself, rather than importing it from the package root

open to suggestions regarding the new file names + catching other import issues

### CODE REVIEW

File: cool_seq_tool/main.py
Comment (korikuzma) [inline cool_seq_tool/main.py:1]: Could rename to just app.py?

**Comment by korikuzma** [comment]
Surprised we haven't had any issues with imports until now. Working on this Variation Normalizer refactor makes me excited for when I can have time to clean up Cool Seq Tool - this PR is a nice start!

File: cool_seq_tool/main.py
Comment (jsstevenson) [reply cool_seq_tool/main.py:1]: @korikuzma is it confusing to have api.py contain the FastAPI `app` object, and `app.py` contain the tools logic?

**Comment by jsstevenson** [comment]
> Surprised we haven't had any issues with imports until now.

I think you'd only find these in some pretty narrow circumstances -- I was trying to stand up some basic docs for my own education and Sphinx's autodoc was unhappy, but otherwise I've never had an issue with them before.

File: cool_seq_tool/data_sources/mane_transcript_mappings.py [hunk 1]
```diff
@@ -4,7 +4,8 @@
 
 import pandas as pd
 
-from cool_seq_tool import MANE_SUMMARY_PATH, logger
+from cool_seq_tool import logger
```
Comment (korikuzma) [inline cool_seq_tool/data_sources/mane_transcript_mappings.py:7]: Did you want to import the logger here? `cool_seq_tool/routers/default.py` and `cool_seq_tool/routers/default.py` also import logger

**Review by korikuzma — APPROVED** [review]

**Comment by korikuzma** [comment]
Oops sorry, was on the wrong branch

**Review by korikuzma — APPROVED** [review]

**Pull request merged**